### DISCUSSION TITLE
how to interact directly with space coordinates?

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1137] [Q&A] [answered]
How can I limit or expand the update scope of objects? I want objects outside the screen at some distance to still update, and objects beyond this area stop updating. I understand how to do this, but I can't figure out how to interact directly with space coordinates

**Comment by rh101** [comment, marked as answer]
What do you mean by "update"?  Are you referring to the `Node::update()` method? 

Check the `void Sprite::draw(Renderer* renderer, const Mat4& transform, uint32_t flags)` function to see how it handles culling, and perhaps you can use the same method to stop updating the object.  

You could also check the position of an object within the world with a to call `getWorldPosition()` on that object (if it's a subclass of `Node`).  You would then do the boundary checks based on that position.

**Reply by 10100010010001001010** [reply]
**What do you mean by "update"?**
I mean "update" like chunks update in minecraft, they are still updating in some area, even tough they are behind the screen.

**You would then do the boundary checks based on that position.**
but if I want to create an object randomly placed on the map? I want to directly tell it the x and y coordinates where it will appear in world, I don't want to depend on the coordinates of another object.

**Reply by rh101** [reply]
> **You would then do the boundary checks based on that position.** but if I want to create an object randomly placed on the map? I want to directly tell it the x and y coordinates where it will appear in world, I don't want to depend on the coordinates of another object.

Why would an object be dependent on the coordinates of another object?  Are you referring to the scene tree (parent/child nodes)? If that is the case, then "world" you're referring to is the base of that tree, which in most cases is a `ax::Scene` object.  Something must exist at the base, which everything else is relative to.

Nothing is stopping you from adding an object to the scene at any X and Y coordinate.  If that object has the scene as the parent, then that is where it will appear in the "world".  If an object is added as a child of another node (not the scene), then it will be relative to that parent node.
